### ISSUE TITLE
If port is given in quotes in the config file, it is silently ignored

### ISSUE THREAD

**Issue by kierenevans** [papertrail/remote_syslog2#84]
Hi,

We encountered an issue when setting the configuration file up with chef whereby if the port number is specified in single quotes, it is silently ignored and the default of 514 is used.

I have submitted a PR to fix the quoted port with the chef cookbook in question ( https://github.com/trappar/remote_syslog2-cookbook/pull/2 ) but thought I should also raise it here.

With a config file like so at `/etc/remote_syslog2.yml`:

```
---
files: []
exclude_files: []
exclude_patterns: []
hostname: foo.example.com
destination:
  host: logs2.papertrailapp.com
  port: '1234'
  protocol: tls
```

and when launched like so: `sudo /usr/local/bin/remote_syslog2 -c /etc/remote_syslog2.yml --pid-file=/var/run/remote_syslog2.pid -D`, the following is outputted:

```
2015-08-11 10:02:24 INFO  remote_syslog.go:114 Connecting to logs2.papertrailapp.com:514 over tls
2015-08-11 10:02:24 CRITICAL  remote_syslog.go:118 Cannot connect to server: dial tcp 173.247.105.170:514: connection refused
```

This immediately fails as logs2.papertrailapp.com isn't open for tcp connections to port 514.

**Comment by troy** [comment]
@kierenevans the result you saw makes sense. The config file is YAML and when it's parsed, a quoted string is treated like a string, even when the quotes contain a number. In this case, it's handled completely differently.

I just commented on that PR and will continue the discussion there.

**Comment by kierenevans** [comment]
This was more a request that the port number be casted to an integer or complained about. 

A scenario:
If the protocol of `tls` is left off, remote_syslog2 defaults to sending over UDP, where the port of 514 is open.
If the public IP of the box was previously assigned to another system, the owner of the previous IP will be able to read the logs sent from this misconfigured daemon.
The owner of the system running this daemon may be unaware of this, as remote_syslog2 has explicitly ignored the configuration for the port.

So, the YAML parser should not swallow errors, especially regarding the port number.

**Comment by eric** [comment]
I agree that we shouldn't be silently ignoring keys if the value type doesn't match what we expect.

In this case of the port, we should definitely be coercing the string into an int. In other cases, if the type is something we don't expect we should either throw an error or at the very least a warning.

**Comment by thegranddesign** [comment]
I really wish this had been fixed.  I just attempted to debug this for over an hour before searching for "connection refused" (which I only saw because I `strace`d it) and found this issue.

Definitely not something that is intuitive.  The general rules to only use integers when you're doing math on them.  Otherwise it shouldn't matter.  Or you should cast it when you need to do something number-y with it.  After all, you don't store a phone number in the database as an integer.  The same logic applies.

**Comment by snorecone** [comment]
Fixed in #170
